K2Bridge/Visitors/HistogramVisitor.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System;
+    using K2Bridge.Models.Request.Aggregations;
+
+    /// <content>
+    /// A visitor for the <see cref="HistogramAggregation"/> element.
+    /// </content>
+    internal partial class ElasticSearchDSLVisitor : IVisitor
+    {
+        /// <inheritdoc/>
+        public void Visit(HistogramAggregation histogramAggregation)
+        {
+            Ensure.IsNotNull(histogramAggregation, nameof(histogramAggregation));
+            EnsureClause.StringIsNotNullOrEmpty(histogramAggregation.Metric, nameof(histogramAggregation.Metric));
+            EnsureClause.StringIsNotNullOrEmpty(histogramAggregation.Field, nameof(histogramAggregation.Field));
+
+            histogramAggregation.KustoQL = $"_data | {KustoQLOperators.Summarize} " + histogramAggregation.SubAggregationsKustoQL +
+            $"{histogramAggregation.Metric} by {EncodeKustoField(histogramAggregation.Key + "%" + histogramAggregation.Keyed)} = ";
```

Comment:
You are creating the same field name 3 times. I'd extract that into a var and use where needed.
`
EncodeKustoField(histogramAggregation.Key + "%" + histogramAggregation.Keyed)
`

Reply:
Done